### DISCUSSION TITLE
Can I use flux-dev commercially?

### DISCUSSION THREAD

**Discussion by jrenzi** [runpod-workers/worker-comfyui#66] [General]
I'm very confused about flux dev licensing. Can I use this worker on runpod to generate images and charge people credits for using it?

**Comment by amirallami-code** [comment]
Yes, you can use it on Runpod and charge people because the AGPL allows commercial use. If you modify the software, you must provide the modified source code to users, as required by the AGPL to ensure software freedom.

**Reply by jrenzi** [reply]
Thanks for your answer. However, I couldn’t find any mention about AGPL in Flux Dev license here https://github.com/black-forest-labs/flux/blob/main/model_licenses/LICENSE-FLUX1-dev

Do you know any source to check for this?

I’m very confused, I seen many sites offering it but I don’t know if they got a license or they are just using at is.

**Reply by amirallami-code** [reply]
Thank you for pointing that out. You're right, I made a mistake mentioning AGPL. After reviewing the Flux Dev license on GitHub, I see it doesn't mention commercial use explicitly. The terms seem restrictive, but without official clarification, it's hard to be certain about allowed usage. For accurate information on commercial use and licensing options, I recommend contacting Black Forest Labs directly. They can provide the most up-to-date and authoritative guidance on how their model can be used.

**Comment by VictorMustin** [comment]
You need a license for commercial use of Flux, unless you run it on a compute provider that has an agreement with BFL (like Replicate or Fal)

**Reply by TimPietrusky** [reply]
Yes, that is correct. You can use the generated images commercially, but you can't sell anyone access to flux-dev on RunPod. You can only sell access to flux-schnell. 

I will see what I can do internally so that we can get this solved at RunPod as well.
